src/gui/FileRevealer.cpp

```diff
@@ -0,0 +1,142 @@
+/*
+ * FileRevealer.cpp - Helper file for cross platform file revealing
+ *
+ * Copyright (c) 2025 Andrew Wiltshire <aw1lt / at/ proton/ dot/me >
+ *
+ * This file is part of LMMS - https://lmms.io
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public
+ * License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program (see COPYING); if not, write to the
+ * Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA 02110-1301 USA.
+ *
+ */
+
+#include "FileRevealer.h"
+
+#include <QDesktopServices>
+#include <QDir>
+#include <QProcess>
+#include <QString>
+#include <QUrl>
+#include <optional>
+
+#include "lmmsconfig.h"
+
+namespace lmms {
+bool FileRevealer::s_canSelect = false;
+
+const QString& FileRevealer::getDefaultFileManager()
+{
+	static std::optional<QString> fileManagerCache;
+	if (fileManagerCache.has_value()) { return fileManagerCache.value(); }
+#if defined(LMMS_BUILD_WIN32)
+	fileManagerCache = "explorer";
+#elif defined(LMMS_BUILD_APPLE)
+	fileManagerCache = "open";
+#else
+	QProcess process;
+	process.start("xdg-mime", {"query", "default", "inode/directory"});
+	process.waitForFinished(3000);
+
+	QString fileManager = QString::fromUtf8(process.readAllStandardOutput()).toLower().trimmed();
+
+	// Handle multiple entries by taking the last non-empty one
+	QStringList fileManagers = fileManager.split(';', Qt::SkipEmptyParts);
+	if (!fileManagers.isEmpty())
+	{
+		fileManager = fileManagers.last(); // Take the last registered manager
```

Comment:
```suggestion
		// The highest priority file manager is last
		fileManager = fileManagers.last();
```
Just to clarify why we're doing it this way